GoogleChromeLabs/ps-analysis-tool
# Feature: Adds IBC compliance & Cookie Set attribute inside CookieDetails Card

## Description


This pull request introduces the IBC (Incrementally Better Cookies) compliance attribute inside the CookieDetails Card component that displays the SameSite attribute of cookies and verifies if their values align with the IBC draft proposal.

The implementation ensures that cookies adhere to the recommended SameSite attribute values, such as "Lax" by default and "None" with the "Secure" flag for cross-site delivery. It provides a clear overview of the compliance status, helping developers verify if their cookies meet the IBC standards.

## Relevant Technical Choices


- Introduces a key-value pair-based UI inside the CookieDetails Card.
  - Adds two new keys: "IBC Compliant" and "Cookie Set" to represent specific attributes.
  - Enables a long hover over each key to display a summary of their use cases, providing contextual information.
- Updates Cookie Headers Parsing Mechanism:
  - Enhances the cookie headers parsing mechanism to streamline the values of a cookie.
  - Takes into account the cookie's header parsed entries, Chrome cookie-store cookie, and previously parsed cookie.
  - Improves the accuracy and consistency of parsed cookie values, ensuring reliable data extraction.
  
## Testing Instructions


- Open a website (say https://indianexpress.com/), now open the Privacy analysis panel in devtools.
- Click on the tab Cookies, and click on any cookie.
- In the CookieDetails card, you can see IBC Compliant and Cookie Set will be visible with relevant values.

## Additional Information:



## Screenshot/Screencast



https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/b5c3f693-ab00-45bf-a4b3-9802eb3dc28f

<img width="795" alt="Screenshot 2023-07-11 at 5 19 56 PM" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/ddde940f-d603-4e1a-bd2c-45ad4fd7fd99">

<img width="794" alt="Screenshot 2023-07-11 at 5 35 01 PM" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/7756337f-cfcd-4500-9c48-9e59786a9632">


---



Fixes #35
